### PR TITLE
Nas 35/layout list of workers

### PR DESCRIPTION
**Pull request by NastyaIgnatchik**

### CODE REVIEW

File: src/pages/profile/Profile.tsx [hunk 1]
```diff
@@ -48,12 +48,19 @@ const Profile = () => {
           lastname: userData.lastname,
           avatar: "123",
         },
-        authContext.token
+        context.token
       )
       .then((response: any) => {
         setIsDisabled(!isDisabled);
+      })
+      .catch(function (error: any) {
```
Comment (ErmolaevAlexey365) [inline src/pages/profile/Profile.tsx:56]: any

**Review by ErmolaevAlexey365 — CHANGES_REQUESTED** [review]

File: src/pages/login/Login.tsx [hunk 2]
```diff
@@ -18,15 +18,10 @@ const Login = () => {
       })
 
       .then((response: any) => {
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/pages/login/Login.tsx:18]: Done

File: src/pages/profile/Profile.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import FormUserData from "../../components/forms/FormUserData";
 import SidebarMenu from "../../components/sidebar/SidebarMenu";
 import { AuthContext } from "../../context/context";
 
-const MainPage = () => {
+const Profile = () => {
   const [userData, setUserData] = useState<any>([{}]);
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/pages/profile/Profile.tsx:8]: done

File: src/utils/dataSorting.ts [hunk 4]
```diff
@@ -28,12 +28,12 @@ export const sortDataInputsAndWorkerCheckbox=(data:any)=>{
...
 
         }
-
-
-
-
-
     }
 
+    // [
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/utils/dataSorting.ts]: не нашла где

File: src/interfaces/interfaces.ts [hunk 5]
```diff
@@ -1,92 +1,118 @@
...
+  sorting: string;
+}
 
+export interface IWorkersData {
+  title: string;
+  id: number;
+  status: string;
+  description: string;
+  removeWorker: () => void;
+  serviceUserAccount: any;
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/interfaces/interfaces.ts:91]: done

File: src/pages/dashboard/Dashboard.tsx [hunk 6]
```diff
@@ -1,61 +1,136 @@
...
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+  const [workerData, setWorkerData] = useState<IWorkersData[]>([]);
+  const [isPasswordModalOpen, setIsPasswordModalOpen] = useState<boolean>(false);
+  const [upworkAccounts, setUpworkAccounts] = useState<string[]>([]);
+  const [accountUserName, setAccountUserName] = useState<string>("");
+  const [accountUserId, setAccountUserId] = useState<number>(0);
+
+  const context = useContext(Context);
+
+  async function submitForm(data: any) {
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/pages/dashboard/Dashboard.tsx:23]: done

File: src/pages/dashboard/Dashboard.tsx [hunk 7]
```diff
@@ -1,61 +1,136 @@
...
+              context.setIsUserLogin(false);
+              context.setIsUserAuth(false);
+          }
+      });
+  }
+
+  async function getWorkers() {
+    await userService
+      .getAllParsers(context.token)
+      .then((response: any) => {
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/pages/dashboard/Dashboard.tsx:53]: done

File: src/pages/dashboard/Dashboard.tsx [hunk 8]
```diff
@@ -1,61 +1,136 @@
...
+  }, [isModalOpen]);
+
+  useEffect(() => {
+    getWorkers();
+  }, []);
 
+  async function getAccounts() {
+    await userService
+      .getAccountsData(context.token)
+      .then((response: any) => {
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/pages/dashboard/Dashboard.tsx:77]: done

File: src/pages/profile/Profile.tsx (hunk 1 above)
Comment (NastyaIgnatchik) [reply src/pages/profile/Profile.tsx:56]: done

File: src/interfaces/interfaces.ts [hunk 9]
```diff
@@ -1,92 +1,118 @@
...
+  isPasswordModalOpen: boolean;
+  setIsPasswordModalOpen: (isPasswordModalOpen: boolean) => void;
+  accountUserName: string;
+  accountUserId: number;
+}
 
+export interface IScratchFormProps {
+  clickHandlerForCloseModal: (e: React.MouseEvent<HTMLButtonElement>) => void;
+  isModalOpen: boolean;
+  submitForm: (data: any) => void;
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/interfaces/interfaces.ts:44]: done

File: src/components/service/userInstance.ts [hunk 10]
```diff
@@ -92,4 +92,70 @@ export const userService = {
       return error.response;
     }
   },
-};
+  async startParser(body: any, token: string) {
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/service/userInstance.ts:95]: done

File: src/components/lists/Workers.tsx [hunk 11]
```diff
@@ -1,14 +1,217 @@
...
+  async function getJobs() {
+    await userService
+      .getParsersById(1, id, context.token)
+      .then((response: any) => {
+          localStorage.setItem(`${id}`, JSON.stringify(response.data.jobs));
+          setJob(response.data.jobs);
+          setStartWorkerLoading(false);
+          getWorkers();
+      })
+      .catch(function (error: any) {
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/lists/Workers.tsx:99]: done

File: src/components/lists/Workers.tsx [hunk 12]
```diff
@@ -1,14 +1,217 @@
...
+    await userService
+      .stopParser(
+        {
+          serviceUserAccountId: accountId,
+          parserId: id,
+          companyUserId: 1,
+        },
+        context.token
+      )
+      .then((response: any) => {
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/lists/Workers.tsx:72]: done

File: src/components/lists/Workers.tsx [hunk 13]
```diff
@@ -1,14 +1,217 @@
...
+        getWorkers();
+      })
+      .catch(function (error: any) {
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/lists/Workers.tsx:75]: done

File: src/components/lists/Workers.tsx [hunk 14]
```diff
@@ -1,14 +1,217 @@
...
+          serviceUserAccountId: accountId,
+          parserId: id,
+          companyUserId: 1,
+        },
+        context.token
+      )
+      .then((response: any) => {
+         getJobs();
+      })
+      .catch(function (error: any) {
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/lists/Workers.tsx:47]: done

File: src/components/lists/Workers.tsx [hunk 15]
```diff
@@ -1,14 +1,217 @@
...
+    await userService
+      .startParser(
+        {
+          serviceUserAccountId: accountId,
+          parserId: id,
+          companyUserId: 1,
+        },
+        context.token
+      )
+      .then((response: any) => {
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/lists/Workers.tsx:44]: done

File: src/components/lists/UserAccount.tsx [hunk 16]
```diff
@@ -14,18 +12,22 @@ const UserAccount = ({
...
     getAccounts();
   }
-    const authContext = useContext(AuthContext);
 
   async function removeUserAccount(id: number) {
     await userService
-      .deleteAccountsData(id, id,authContext.token)
+      .deleteAccountsData(id, id, context.token)
       .then((response: any) => {})
       .catch(function (error: any) {
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/lists/UserAccount.tsx:26]: done

File: src/components/lists/UserAccount.tsx [hunk 17]
```diff
@@ -14,18 +12,22 @@ const UserAccount = ({
...
     removeUserAccount(id);
     getAccounts();
   }
-    const authContext = useContext(AuthContext);
 
   async function removeUserAccount(id: number) {
     await userService
-      .deleteAccountsData(id, id,authContext.token)
+      .deleteAccountsData(id, id, context.token)
       .then((response: any) => {})
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/lists/UserAccount.tsx:25]: done

File: src/components/lists/Jobs.tsx [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+              onClick={() => setViewMore(!viewMore)}
+            />
+          </>
+        )
+      ) : (
+        <div className={styles.skills}>
+          {" "}
+          {skills.split(",").map((el: string) => {
+            if (el != "")
+              return <Chip label={el} size="small" color="primary" />;
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/lists/Jobs.tsx:92]: done

File: src/components/lists/Jobs.tsx [hunk 19]
```diff
@@ -0,0 +1,107 @@
...
+              onClick={() => setViewMore(!viewMore)}
+            />
+          </>
+        ) : (
+          <>
+            <div className={styles.skills}>
+              {" "}
+              {skills.split(",").map((el: string) => {
+                if (el != "")
+                  return <Chip label={el} size="small" color="primary" />;
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/lists/Jobs.tsx:76]: done

File: src/components/lists/Jobs.tsx [hunk 20]
```diff
@@ -0,0 +1,107 @@
...
+      )}
+
+      {skills.split(",").length > 11 ? (
+        !viewMore ? (
+          <>
+            <div className={styles.skills_close}>
+              {" "}
+              {skills.split(",").map((el: string) => {
+                if (el != "")
+                  return <Chip label={el} size="small" color="primary" />;
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/lists/Jobs.tsx:60]: done

File: src/components/forms/worker/WorkerForm.tsx [hunk 21]
```diff
@@ -380,33 +321,27 @@ const WorkerForm = ({ clickHandlerForCloseModal,isModalOpen,submitForm }: IScrat
...
             <FormControlLabel
               label="Intermediate"
               control={
                 <Controller
                   name="contractor_tier.1"
                   control={control}
-                  render={({ field: { value, onChange } }) => (
+                  render={({ field: {  onChange } }) => (
                     <Checkbox
                       value={"2"}
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/forms/worker/WorkerForm.tsx:346]: done

File: src/components/forms/worker/WorkerForm.tsx [hunk 22]
```diff
@@ -380,33 +321,27 @@ const WorkerForm = ({ clickHandlerForCloseModal,isModalOpen,submitForm }: IScrat
...
-
-                />
-                  }
-
+                <Controller
+                  name="contractor_tier.0"
+                  control={control}
+                  render={({ field: { onChange } }) => (
+                    <Checkbox
+                      value={"1"}
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/forms/worker/WorkerForm.tsx:330]: done

File: src/components/forms/worker/WorkerForm.tsx [hunk 23]
```diff
@@ -312,15 +253,15 @@ const WorkerForm = ({ clickHandlerForCloseModal,isModalOpen,submitForm }: IScrat
                     {...field}
                   >
                     {upworkAccounts.map((option: any) => (
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/forms/worker/WorkerForm.tsx:255]: done

File: src/components/forms/worker/WorkerForm.tsx [hunk 24]
```diff
@@ -16,40 +16,31 @@ import {
...
+const WorkerForm = ({ clickHandlerForCloseModal,submitForm,upworkAccounts }: IScratchFormProps) => {
   const [isChecked, setIsChecked] = useState<boolean>(false);
   const [isLocationData, setIsLocationData] = useState<any[]>([]);
   const [isHourly, setIsHourly] = useState<boolean>(false);
   const [isFixedPrice, setIsFixedPrice] = useState<boolean>(false);
   const [hourlyNumberMin, setHourlyNumberMin] = useState<any>("");
   const [hourlyNumberMax, setHourlyNumberMax] = useState<any>("");
-
   const [FixedPriceNumberMin, setFixedPriceNumberMin] = useState<any>("");
   const [FixedPriceNumberMax, setFixedPriceNumberMax] = useState<any>("");
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/forms/worker/WorkerForm.tsx:37]: done

File: src/components/forms/worker/WorkerForm.tsx [hunk 25]
```diff
@@ -16,40 +16,31 @@ import {
...
-const WorkerForm = ({ clickHandlerForCloseModal,isModalOpen,submitForm }: IScratchFormProps) => {
+
+
+
+const WorkerForm = ({ clickHandlerForCloseModal,submitForm,upworkAccounts }: IScratchFormProps) => {
   const [isChecked, setIsChecked] = useState<boolean>(false);
   const [isLocationData, setIsLocationData] = useState<any[]>([]);
   const [isHourly, setIsHourly] = useState<boolean>(false);
   const [isFixedPrice, setIsFixedPrice] = useState<boolean>(false);
   const [hourlyNumberMin, setHourlyNumberMin] = useState<any>("");
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/forms/worker/WorkerForm.tsx:34]: done

File: src/components/buttons/verticalMenu/VertMenu.tsx [hunk 26]
```diff
@@ -0,0 +1,62 @@
...
+        open={open}
+        onClose={handleClose}
+        PaperProps={{
+          style: {
+            maxHeight: ITEM_HEIGHT * 4.5,
+            width: "20ch",
+          },
+        }}
+      >
+        {options.map((option: any) => (
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/buttons/verticalMenu/VertMenu.tsx:52]: done

File: src/components/buttons/verticalMenu/VertMenu.tsx [hunk 27]
```diff
@@ -0,0 +1,62 @@
...
+
+const VertMenu = ({ options, removeWorker }: any) => {
+  const ITEM_HEIGHT = 48;
+
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = (option: any) => {
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/buttons/verticalMenu/VertMenu.tsx:15]: done

File: src/components/buttons/verticalMenu/VertMenu.tsx [hunk 28]
```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import IconButton from "@mui/material/IconButton";
+import Menu from "@mui/material/Menu";
+import MenuItem from "@mui/material/MenuItem";
+import MoreVertIcon from "@mui/icons-material/MoreVert";
+
+const VertMenu = ({ options, removeWorker }: any) => {
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/buttons/verticalMenu/VertMenu.tsx:7]: done

File: src/pages/dashboard/Dashboard.tsx [hunk 29]
```diff
@@ -1,19 +1,59 @@
...
+import {AuthContext} from "../../context/context";
+import {IWorkerData} from "../../interfaces/interfaces";
 
 
 const Dashboard = () => {
 const [isModalOpen,setIsModalOpen]=useState<boolean>(false);
+const [workerData, setWorkerData] = useState<IWorkerData[]>([])
+
+    const authContext = useContext(AuthContext);
+    async function submitForm (data: any) {
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/pages/dashboard/Dashboard.tsx]: done

File: src/interfaces/interfaces.ts [hunk 30]
```diff
@@ -53,7 +58,7 @@ id:number;
 
 
  export interface IWorkerData{
-     timer:number;
+     timer:any;
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/interfaces/interfaces.ts]: done

File: src/interfaces/interfaces.ts [hunk 31]
```diff
@@ -28,10 +28,15 @@ export interface IPropsForFormLogin{
 export interface IOpenModalButton{
     isModalOpen:boolean;
     setIsModalOpen:(isModalOpen:boolean)=>void;
+    submitForm: (data:any)=>void;
+
 }
 export interface IScratchFormProps{
     clickHandlerForCloseModal:(e: React.MouseEvent<HTMLButtonElement>)=>void;
     isModalOpen:boolean;
+    submitForm: (data:any)=>void;
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/interfaces/interfaces.ts]: done

File: src/interfaces/interfaces.ts [hunk 32]
```diff
@@ -28,10 +28,15 @@ export interface IPropsForFormLogin{
 export interface IOpenModalButton{
     isModalOpen:boolean;
     setIsModalOpen:(isModalOpen:boolean)=>void;
+    submitForm: (data:any)=>void;
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/interfaces/interfaces.ts]: done

File: src/components/schemas/Schema.ts [hunk 33]
```diff
@@ -12,7 +12,7 @@ export const schema = yup
 
 export const schemaWorker = yup
   .object({
-    timer: yup.number().required("This is a required field"),
+    // timer: yup.number().required("This is a required field"),
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/schemas/Schema.ts]: done

File: src/components/forms/worker/WorkerForm.tsx [hunk 34]
```diff
@@ -228,7 +230,9 @@ const WorkerForm = ({ clickHandlerForCloseModal,isModalOpen }: IScratchFormProps
       <Typography variant="h5" component="h4">
         Add worker
       </Typography>
-      <form onSubmit={handleSubmit(submitForm)}>
+      <form onSubmit={handleSubmit((data:any)=>{
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/forms/worker/WorkerForm.tsx:177]: done

File: src/interfaces/interfaces.ts [hunk 35]
```diff
@@ -31,6 +31,7 @@ export interface IOpenModalButton{
 }
 export interface IScratchFormProps{
     clickHandlerForCloseModal:(e: React.MouseEvent<HTMLButtonElement>)=>void;
+    isModalOpen:boolean;
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/interfaces/interfaces.ts:33]: done

File: src/components/forms/worker/WorkerForm.tsx [hunk 36]
```diff
@@ -97,20 +101,39 @@ const WorkerForm = ({ clickHandlerForCloseModal,isModalOpen }: IScratchFormProps
       });
   }
 
+
+
   useEffect(() => {
+    // isCheckboxChecked()
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/forms/worker/WorkerForm.tsx]: done

File: src/components/forms/worker/WorkerForm.tsx [hunk 37]
```diff
@@ -97,20 +101,39 @@ const WorkerForm = ({ clickHandlerForCloseModal,isModalOpen }: IScratchFormProps
...
   const falseOrValue = (e: React.ChangeEvent<HTMLInputElement>) => {
-    if (!e.target.checked) {
+    if (!e.target.checked ) {
       return (e.target.value = "false");
     } else {
       return e.target.value;
     }
   };
 
+ // function isCheckboxChecked(){
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/forms/worker/WorkerForm.tsx]: done

File: src/components/forms/worker/WorkerForm.tsx [hunk 38]
```diff
@@ -269,7 +277,7 @@ const WorkerForm = ({ clickHandlerForCloseModal,isModalOpen }: IScratchFormProps
               <FormHelperText
                 style={{ background: "#e5f4ff", margin: "0", color: "#d32f2f" }}
               >
-                {errors.timer?.message}
+                {/*{errors.timer?.message}*/}
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/forms/worker/WorkerForm.tsx]: done

File: src/components/forms/worker/WorkerForm.tsx [hunk 39]
```diff
@@ -312,7 +320,7 @@ const WorkerForm = ({ clickHandlerForCloseModal,isModalOpen }: IScratchFormProps
...
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/forms/worker/WorkerForm.tsx]: done

File: src/components/forms/worker/WorkerForm.tsx [hunk 40]
```diff
@@ -372,20 +380,26 @@ const WorkerForm = ({ clickHandlerForCloseModal,isModalOpen }: IScratchFormProps
...
-                    />
-                  )}
+                <Checkbox
+                    {...register("contractor_tier.0")}
+                    value={"1"}
+                    defaultChecked={false}
+
+
+
+                    // checked={isCheckboxesChecked}
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/forms/worker/WorkerForm.tsx]: done

File: src/components/lists/Workers.tsx [hunk 41]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import styles from "../../styles/dashboard/dashboard.module.css";
+import {IWorkersProps} from "../../interfaces/interfaces";
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/lists/Workers.tsx]: done

File: src/components/buttons/menuButton/MenuButton.tsx [hunk 42]
```diff
@@ -1,18 +1,18 @@
 import React, {useContext} from "react";
 import styles from "../../../styles/profile/profile.module.css";
 import MenuIcon from "@mui/icons-material/Menu";
-
-import {AuthContext} from "../../../context/context";
+import {Context} from "../../../context/context";
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/buttons/menuButton/MenuButton.tsx:4]: done

File: src/components/forms/worker/WorkerForm.tsx [hunk 43]
```diff
@@ -422,7 +357,7 @@ const WorkerForm = ({ clickHandlerForCloseModal,isModalOpen,submitForm }: IScrat
                 <Controller
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/forms/worker/WorkerForm.tsx:357]: done

File: src/components/modals/WorkerModal.tsx [hunk 44]
```diff
@@ -3,7 +3,8 @@ import {IOpenModalButton} from "../../interfaces/interfaces";
 import styles from '../../styles/dashboard/dashboard.module.css';
 import CloseIcon from "@mui/icons-material/Close";
 
-import WorkerFormComponents from "../forms/worker/WorkerForm";
+
+import WorkerForm from "../forms/worker/WorkerForm";
 
 
 const WorkerModal = ({  isModalOpen,setIsModalOpen }:IOpenModalButton) => {
```
[the remark this answers is not in the document]
Comment (NastyaIgnatchik) [reply src/components/modals/WorkerModal.tsx]: done

**Review by ErmolaevAlexey365 — APPROVED** [review]

**Pull request merged**